news/_posts/2020-02-04-the-new-prestashop-test-framework.md

```diff
@@ -0,0 +1,96 @@
+This campaign is the biggest so far, and my team and I are proud to say that it works wonderfully. There are still some errors sometimes but we hope to iron things out with time. It already proved itself useful by finding bugs ! It’s also very consistent: when it fails it’s always in the same spot and for the same reason.
+
+The **end-to-end campaign** will cover what’s left: it is used to test real user paths. The goal is to make sure all the actions a merchant can do are covered by tests, so that the Tech team can be confident about the level of quality of the application. The Product team (and the QA team) already have a good idea of those user paths (gathered through the interviews with merchants) and we will try to implement paths that are the most relevant to a merchant daily and weekly recurring actions.
+
+There is one last campaign: we want to make sure old issues stay fixed and don’t come back after a while. There will be a regression campaign for this sole purpose after (or at least get up to a comfortable coverage) the main campaign is finished. It will target critical and major issues for the last few minor versions.
+
+
+## That’s nice, but where can I see all the action ?
+
+Our [nightly board is public](https://nightly.prestashop.com/)! You can check it and open every report. The current version was quickly engineered to let us gather the data and present it in an acceptable way, but it’s being redone (thanks to the UX team and our front-end developer!) in a few weeks (hopefully), using the UI Kit developed for PrestaShop. Keep an eye on it!
+
+All the tests I present in this article are also available in the [develop](https://github.com/PrestaShop/PrestaShop/tree/develop/tests/puppeteer) and [1.7.7.x](https://github.com/PrestaShop/PrestaShop/tree/1.7.7.x/tests/puppeteer) branches in the GitHub repository. Feel free to explore and execute them.
+
+Don’t forget that you can help us by [writing tests or Page Objects through Pull Requests](https://devdocs.prestashop.com/1.7/testing/how-to-create-your-own-web-acceptance-tests/#creating-a-web-acceptance-test)!
+
+Addendum: the team behind Puppeteer recently revealed their new project: [PlayWright](https://github.com/microsoft/playwright). Based on their work on Puppeteer, it’s also a Node library but able to communicate with Chromium, Firefox, and Edge. As soon as they release a stable version, we’ll look seriously into it, since testing all major browsers would be a nice addition.
+
+
+## TL;DR
+
+The PrestaShop QA team is dropping Selenium and redoing the whole Core test campaign using another tool called Puppeteer, with another design pattern (Page Object Model).
+We aim for better coverage, faster answers for the devs, less time lost in maintaining and fixing the tests, and overall a real change in the quality of the open source project we love !
+
+As always, at PrestaShop we’re impatient to hear your comments and suggestions.
+
```

Comment:
Might be more useful if put at the top of the article 😉

Reply:
Why not. But this is very common to find the TLDR at the bottom for tech articles.

Reply:
I replaced it at the top of the article, but I agree with you @ttoine i've seen it at the bottom more frequently than at the top.